src/ImageIO/CGImageAnimation.cs

```diff
@@ -0,0 +1,141 @@
+        static extern int CGAnimateImageDataWithBlock ( /* CFDataRef _Nonnull */ IntPtr data, /* CFDictionaryRef _Nullable */ IntPtr options, /* CGImageSourceAnimationBlock _Nonnull */ ref BlockLiteral block);
+
+        [Introduced (PlatformName.MacOSX, 10, 15, PlatformArchitecture.All)]
+        [Introduced (PlatformName.iOS, 13, 0, PlatformArchitecture.All)]
+        [BindingImpl (BindingImplOptions.Optimizable)]
+        public int AnimateImage (NSUrl url, NSDictionary options, [BlockProxy (typeof (NIDCGImageSourceAnimationBlock))] CGImageSourceAnimationBlock block)
+        {
+#if IOS && ARCH_32
+            throw new PlatformNotSupportedException ();
+#else
+            if (url == null)
+                throw new ArgumentNullException (nameof (url));
+            if (block == null)
+                throw new ArgumentNullException (nameof (block));
+
+            int ret;
+            BlockLiteral block_block;
+            block_block = new BlockLiteral ();
+            block_block.SetupBlockUnsafe (SDCGImageSourceAnimationBlock.Handler, block);
+
+            try {
+                ret = CGAnimateImageAtURLWithBlock (url.Handle, options.GetHandle (), ref block_block);
+            } finally {
+                block_block.CleanupBlock ();
+            }
+
+            return ret;
+#endif
+        }
+
+        [Introduced (PlatformName.MacOSX, 10, 15, PlatformArchitecture.All)]
+        [Introduced (PlatformName.iOS, 13, 0, PlatformArchitecture.All)]
+        [BindingImpl (BindingImplOptions.Optimizable)]
+        public int AnimateImage (NSData data, NSDictionary options, [BlockProxy (typeof (NIDCGImageSourceAnimationBlock))] CGImageSourceAnimationBlock block)
+        {
+#if IOS && ARCH_32
+            throw new PlatformNotSupportedException ();
+#else
+            if (data == null)
+                throw new ArgumentNullException (nameof (data));
+            if (block == null)
+                throw new ArgumentNullException (nameof (block));
+
+            int ret;
+            BlockLiteral block_block;
+            block_block = new BlockLiteral ();
+            block_block.SetupBlockUnsafe (SDCGImageSourceAnimationBlock.Handler, block);
+
+            try {
+                ret = CGAnimateImageDataWithBlock (data.Handle, options.GetHandle (), ref block_block);
+            } finally {
+                block_block.CleanupBlock ();
+            }
+
+            return ret;
+#endif
+        }
+
+        //
+        // This class bridges native block invocations that call into C#
+        //
+        static internal class SDCGImageSourceAnimationBlock
+        {
+            static internal readonly DCGImageSourceAnimationBlock Handler = Invoke;
+
+            [MonoPInvokeCallback (typeof (DCGImageSourceAnimationBlock))]
+            static void Invoke (IntPtr block, nint index, IntPtr image, [System.Runtime.InteropServices.MarshalAs (System.Runtime.InteropServices.UnmanagedType.I1)] out bool stop)
```

Comment:
`System.Runtime.InteropServices` is already in your `using` clauses so it can be removed.
I'm lazy and rather not scroll horizontally when reviewing ;-)